media/server/gstplayer/source/tasks/webAudio/SetCaps.cpp

```diff
@@ -0,0 +1,156 @@
+{
+public:
+    WebAudioCapsBuilder(std::shared_ptr<IGstWrapper> gstWrapper, std::shared_ptr<IGlibWrapper> glibWrapper)
+        : m_gstWrapper(gstWrapper), m_glibWrapper(glibWrapper)
+    {
+    }
+    virtual GstCaps *buildCaps() = 0;
+
+protected:
+    std::shared_ptr<IGstWrapper> m_gstWrapper;
+    std::shared_ptr<IGlibWrapper> m_glibWrapper;
+};
+
+class WebAudioPcmCapsBuilder : public WebAudioCapsBuilder
+{
+public:
+    WebAudioPcmCapsBuilder(std::shared_ptr<IGstWrapper> gstWrapper, std::shared_ptr<IGlibWrapper> glibWrapper,
+                           const WebAudioPcmConfig &pcmConfig)
+        : WebAudioCapsBuilder(gstWrapper, glibWrapper), m_pcmConfig(pcmConfig)
+    {
+    }
+
+    GstCaps *buildCaps() override
+    {
+        GstCaps *caps = m_gstWrapper->gstCapsNewEmptySimple("audio/x-raw");
+        addFormat(caps);
+        m_gstWrapper->gstCapsSetSimple(caps, "channels", G_TYPE_UINT, m_pcmConfig.channels, "layout", G_TYPE_STRING,
+                                       "interleaved", "rate", G_TYPE_UINT, m_pcmConfig.rate, nullptr);
+
+        return caps;
+    }
+
+protected:
+    void addFormat(GstCaps *caps)
+    {
+        std::string format;
+
+        if (m_pcmConfig.isFloat)
+        {
+            format += "F";
+        }
+        else if (m_pcmConfig.isSigned)
+        {
+            format += "S";
+        }
+        else
+        {
+            format += "U";
+        }
+
+        format += std::to_string(m_pcmConfig.sampleSize);
+
+        if (m_pcmConfig.isBigEndian)
+        {
+            format += "BE";
+        }
+        else
+        {
+            format += "LE";
+        }
+
+        m_gstWrapper->gstCapsSetSimple(caps, "format", G_TYPE_STRING, format.c_str(), nullptr);
+    }
+
+    const WebAudioPcmConfig &m_pcmConfig;
+};
+}; // namespace
+
+SetCaps::SetCaps(WebAudioPlayerContext &context, std::shared_ptr<IGstWrapper> gstWrapper,
+                 std::shared_ptr<IGlibWrapper> glibWrapper, const std::string &audioMimeType,
+                 const WebAudioConfig *config)
+    : m_context{context}, m_gstWrapper{gstWrapper}, m_glibWrapper{glibWrapper}, m_audioMimeType{audioMimeType}
+{
+    RIALTO_SERVER_LOG_DEBUG("Constructing SetCaps");
+
+    if (m_audioMimeType == "audio/x-raw")
+    {
+        m_config.pcm = config->pcm;
+    }
+}
+
+SetCaps::~SetCaps()
+{
+    RIALTO_SERVER_LOG_DEBUG("SetCaps finished");
+}
+
+void SetCaps::execute() const
+{
+    RIALTO_SERVER_LOG_DEBUG("Executing SetCaps");
+
+    GstCaps *caps = createCapsFromMimeType();
+    gchar *capsStr = m_gstWrapper->gstCapsToString(caps);
+    std::string strCaps = capsStr;
+    m_glibWrapper->gFree(capsStr);
+
+    RIALTO_SERVER_LOG_DEBUG("caps str: '%s'", strCaps.c_str());
+
+    GstCaps *appsrcCaps = m_gstWrapper->gstAppSrcGetCaps(GST_APP_SRC(m_context.source));
+    if ((!appsrcCaps) || (!m_gstWrapper->gstCapsIsEqual(appsrcCaps, caps)))
+    {
+        RIALTO_SERVER_LOG_MIL("Updating web audio appsrc caps to '%s'", strCaps.c_str());
```

Comment:
Should this be _DEBUG?

Reply:
We print a Milstone log when setting the caps in MediaPipeline so i did the same here, happy to change it to Debug though